src/main.ts

```diff
@@ -234,6 +234,9 @@ app.on("ready", async () => {
     debug("ready");
     initApp();
 
+    globalShortcut.register("CommandOrControl+w", () => {
```

Comment:
The proposed solution is incorrect. `globalShortcut` is for "global" shortcuts :)
(system-wide)
See:
https://electronjs.org/docs/api/global-shortcut

The solution is to re-instate the typical "window" menu which has built-in accelerator shortuts, i.e. `role: "close"`, `role: "front"`, `role: "minimize"`, `role: "togglefullscreen"`, etc. and for MacOS (darwin) also in the first "about" menu `role: "services"`, `role: "hide"`, `role: "hideothers"`, `role: "unhide"`, `role: "quit"`.

See:
https://electronjs.org/docs/api/menu#examples